crates/interpreter/src/value.rs

```diff
@@ -71,3 +92,18 @@ pub fn strict_equality(x: &JSValue, y: &JSValue) -> bool {
         _ => false,
     }
 }
+
+pub fn equality(x: &JSValue, y: &JSValue) -> bool {
+    match (x, y) {
+        (JSValue::Undefined, b) => is_nullish(b),
+        (JSValue::Null, b) => is_nullish(b),
+        (JSValue::Boolean(a), b) => a == &to_boolean(b),
+        (JSValue::Number(a), b) => a == &to_number(b),
+        (JSValue::String(ref a), b) => a == &to_string(b),
+        // TODO: fix this
+        (JSValue::Object(ref a), JSValue::Object(ref b)) => a.as_ptr() == b.as_ptr(),
+        // TODO: fix this
+        (JSValue::NativeFunction(a), JSValue::NativeFunction(b)) => std::ptr::eq(a, b),
+        _ => false,
+    }
```

Comment:
these seem not to follow https://tc39.es/ecma262/#sec-abstract-equality-comparison
both about the order and conditions.